src/openzeppelin/tests/test_accesscontrol.cairo

```diff
@@ -2,124 +2,308 @@ use openzeppelin::access::accesscontrol::AccessControl;
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    assert(AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should be granted');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_grant_role_multiple_times_for_granted_role() {
+    setup();
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    assert(AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should still be granted');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_grant_role_already_granted() {
+    setup();
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    assert(AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should still be held');
 }
 
 #[test]
 #[available_gas(2000000)]
 #[should_panic(expected: ('Caller is missing role', ))]
 fn test_grant_role_unauthorized() {
     setup();
-    set_caller_address(ACCOUNT2());
-    AccessControl::grant_role(DEFAULT_ADMIN_ROLE, ACCOUNT2());
+    set_caller_address(AUTHORIZED());
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+}
+
+//
+// revoke_role
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_revoke_role_for_role_not_granted() {
+    setup();
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should be revoked');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_revoke_role_for_granted_role() {
+    setup();
+    AccessControl::grant_role(ROLE, AUTHORIZED());
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should be revoked');
 }
 
 #[test]
 #[available_gas(2000000)]
-fn test_revoke_role() {
+fn test_revoke_role_multiple_times_for_granted_role() {
     setup();
-    AccessControl::grant_role(DEFAULT_ADMIN_ROLE, ACCOUNT2());
+    AccessControl::grant_role(ROLE, AUTHORIZED());
 
-    set_caller_address(ACCOUNT2());
-    AccessControl::revoke_role(DEFAULT_ADMIN_ROLE, ACCOUNT1());
-    assert(!AccessControl::has_role(DEFAULT_ADMIN_ROLE, ACCOUNT1()), 'Role should be revoked');
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should still be revoked');
 }
 
 #[test]
 #[available_gas(2000000)]
 #[should_panic(expected: ('Caller is missing role', ))]
 fn test_revoke_role_unauthorized() {
     setup();
-    set_caller_address(ACCOUNT2());
-    AccessControl::revoke_role(DEFAULT_ADMIN_ROLE, ACCOUNT1());
+
+    set_caller_address(OTHER());
+    AccessControl::revoke_role(ROLE, AUTHORIZED());
+}
+
+//
+// renounce_role
+//
+
+#[test]
+#[available_gas(2000000)]
+fn test_renounce_role_for_role_not_granted() {
+    setup();
+    set_caller_address(AUTHORIZED());
+
+    AccessControl::renounce_role(ROLE, AUTHORIZED());
+    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should be renounced');
```

Comment:
```suggestion
    assert(!AccessControl::has_role(ROLE, AUTHORIZED()), 'Role should not be held');
```
or something similar. it might confuse since the intent is not to *renounce*, but to not revert. as the comment above, we might not even need the assert.

Reply:
we should either assume the role is not granted (i lean towards this), or at most assert initial state validity by checking the role is not granted **before** calling`renounce_role`. what does the assert after add?

currently it seems we're testing that `renounce_role` does not grant the role

Reply:
Agreed in assuming the role is not granted and the assertion is not needed